integrations/influxdb_metrics_dashboard/lambda/upload_dashboard/main.go

```diff
@@ -0,0 +1,459 @@
+		bytes.NewBuffer(jsonDashboard),
+		serviceAccountTokenKey,
+	)
+	if err != nil {
+		return "", err
+	}
+	uploadGrafanaDashboardResp, err := sendGrafanaHttpReq(*httpClient, uploadGrafanaDashboardReq)
+	if err != nil {
+		log.Printf("Failed to upload Grafana dashboard: %s", err)
+		return "", err
+	}
+
+	if uploadGrafanaDashboardResp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("error status code returned from uploading dashboard: %d", uploadGrafanaDashboardResp.StatusCode)
+	}
+
+	return "Dashboard created successfully", nil
+}
+
+func createGrafanaDashboard() (string, error) {
+	const SleepDuration = 5
+	const MaxWaitIntervals = 200
+
+	workspaceName := os.Getenv("GrafanaWorkspaceName")
+	if workspaceName == "" {
+		return "", fmt.Errorf("Failed to get GrafanaWorkspaceName environment variable")
+	}
+	datasourceName := os.Getenv("TimestreamDatasourceName")
+	if datasourceName == "" {
+		return "", fmt.Errorf("Failed to get TimestreamDatasourceName environment variable")
+	}
+	dashboardName := os.Getenv("DashboardName")
+	if dashboardName == "" {
+		return "", fmt.Errorf("Failed to get DashboardName environment variable")
+	}
+	databaseName := os.Getenv("DatabaseName")
+	if databaseName == "" {
+		return "", fmt.Errorf("Failed to get DatabaseName environment variable")
+	}
+
+	awsConfig, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		log.Printf("error loading default AWS config: %s", err)
+		return "", err
+	}
+
+	grafanaServiceAccountTokenName := "ADMIN"
+	var grafanaServiceAccountTokenSecondsToLive int32 = 86400 //24 hours
+	grafanaClient := grafana.NewFromConfig(awsConfig)
+
+	serviceAccountTokenKey := ""
+	grafanaWorkspace, err := getWorkspaceByName(grafanaClient, workspaceName)
+	if err != nil {
+		log.Printf("Failed to get workspace: %s", err)
+		return "", err
+	}
+
+	waiterInterval := 0
+
+	if grafanaWorkspace.Status != grafanaTypes.WorkspaceStatusActive {
+		for waiterInterval < MaxWaitIntervals {
+			grafanaWorkspace, err = getWorkspaceByName(grafanaClient, workspaceName)
+			if err != nil {
+				log.Printf("Failed to get workspace: %s", err)
+				return "", err
+			}
+			if grafanaWorkspace.Status == grafanaTypes.WorkspaceStatusActive {
+				break
+			}
+			waiterInterval++
+			time.Sleep(SleepDuration * time.Second)
+		}
+	}
+
+	if waiterInterval == MaxWaitIntervals {
+		return "", fmt.Errorf("Failed to create service account due to workspace creation timeout")
+	}
+
+	serviceAccountOutput, err := grafanaClient.CreateWorkspaceServiceAccount(
+		context.TODO(),
+		&grafana.CreateWorkspaceServiceAccountInput{
+			GrafanaRole: grafanaTypes.RoleAdmin,
+			Name:        &workspaceName,
+			WorkspaceId: grafanaWorkspace.Id,
+		},
+	)
+	if err != nil {
+		log.Printf("Error creating workspace service account: %s", err)
+		return "", err
+	}
+	serviceAccountTokenOutput, err := grafanaClient.CreateWorkspaceServiceAccountToken(context.TODO(), &grafana.CreateWorkspaceServiceAccountTokenInput{
```

Comment:
Did you have any problems with creating a service account token when one already exists? In the Lambda sample application, I had to make it find the existing token and delete it before creating a new one.

Reply:
I tested and there was an issue if a service account token already exists. I have resolved the issue and now delete the token if one exists.